File: ProjectEuler/Problems/Problem036.cs
using Microsoft.Extensions.Logging;
using System;

namespace ProjectEuler.Problems
{
    public class Problem036 : Problem<long>
    {
        public Problem036(ILogger<Problem036> logger) : base(logger) { }

        protected override long CalculateAnswer()
        {
            var answer = 0L;
            var n = 1000000L;

            Logger.LogDebug("Incremental Results:");
            for (var i = 1L; i < n; ++i)
            {
                var iAsBinary = Convert.ToString(i, 2);
                if (i.IsPalindrome() && iAsBinary.IsPalindrome())
                {
                    Logger.LogDebug("{i}; {iAsBinary}", i, iAsBinary);
                    answer += i;
                }
            }

            return answer;
        }
    }
}

File: ProjectEulerPython/problems/problem_036.py
import int_helpers
import str_helpers
from problems.problem import Problem

class Problem036(Problem):

  def calculate_answer(self) -> int:
    answer = 0
    n = 1000000

    self.print_detail("Incremental Results:")

    for i in range(1, n):
      i_bin = bin(i)[2:]
      if int_helpers.is_palindrome(i) and str_helpers.is_palindrome(i_bin):
        self.print_detail(f"{i}; {i_bin}")
        answer += i

    return answer